### ISSUE TITLE
http/2 hyper errors in local development

### ISSUE THREAD

**Issue by benburkert** [fastly/Viceroy#310]
Hello, I'm using viceroy on my local machine to test an edge component written in go (using `go1.21.1`, not `tinygo`). I updated my viceroy version from `0.7.0` to `0.8.0` and I'm now seeing the following errors in my test runs:

```
Error: hyper::Error(Http2, Error { kind: GoAway(b"", FRAME_SIZE_ERROR, Library) })
2023-09-16T17:14:59.089823Z ERROR Hostcall yielded an error: http2 error: connection error detected: frame with invalid size
```

This is for a backend service which is not using HTTPS or HTTP/2 in local development.

**Comment by benburkert** [comment]
Some extra context, when I run `fastly compute serve` with an HTTP/1.1 backend over HTTPS (no http/2 support), I see the following output:

```
fastly       | 2023-09-16T20:47:22.574139Z  INFO request{id=0}: handling request GET http://anchor.lcl.host:44305/
fastly       | 2023-09-16T20:47:22.583846Z  WARN Unexpected; request h2 for grpc, but got nothing back from ALPN
web          | Unsupported HTTP method used: PRI
fastly       | Error: hyper::Error(Http2, Error { kind: GoAway(b"", FRAME_SIZE_ERROR, Library) })
fastly       | 2023-09-16T20:47:22.584730Z ERROR request{id=0}: Hostcall yielded an error: http2 error: connection error detected: frame with invalid size
```

The `fastly` lines are from viceroy, the `web` line is from the backend server.

**Comment by dkegel-fastly** [comment]
I'm seeing this, too.

**Comment by jagobagascon** [comment]
I'm having the same problem with Go `1.21.1` and the latest Fastly CLI. For anyone having the same issue, you can downgrade viceroy to 0.7.0 by adding this to the `fastly.toml` file:
```toml
[local_server]
  viceroy_version = "0.7.0"
```

**Comment by acw** [comment]
Hello everyone! I think I've addressed the underlying issue, and released Viceroy 0.8.1 with a fix. Can folks tell me if that resolves the issue?

**Comment by GeeWee** [comment]
Issue seems fixed in 0.8.1 for me.

**Comment by jagobagascon** [comment]
Can confirm, thanks for the quick fix!

**Comment by benburkert** [comment]
Yep, fixed in 0.8.1, thanks!
